pkg/services/ngalert/schedule/schedule.go

```diff
@@ -700,26 +700,11 @@ func (r *alertRuleRegistry) del(key models.AlertRuleKey) (*alertRuleInfo, bool)
 	return info, ok
 }
 
-func (r *alertRuleRegistry) iter() <-chan models.AlertRuleKey {
-	c := make(chan models.AlertRuleKey)
-
-	f := func() {
-		r.mu.Lock()
-		defer r.mu.Unlock()
-
-		for k := range r.alertRuleInfo {
-			c <- k
-		}
-		close(c)
-	}
-	go f()
-
-	return c
-}
-
 func (r *alertRuleRegistry) keyMap() map[models.AlertRuleKey]struct{} {
-	definitionsIDs := make(map[models.AlertRuleKey]struct{})
-	for k := range r.iter() {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	definitionsIDs := make(map[models.AlertRuleKey]struct{}, len(r.alertRuleInfo))
```

Comment:
You'll be a teensy bit faster on concurrent calls if you do this allocation outside of the lock.

Reply:
yea, in this particular case this will not make any difference because this registry is actually used in a synchronous context 😄